src/arithmetic.jl

```diff
@@ -0,0 +1,59 @@
+Base.promote_rule(::Type{Arf}, ::Type{<:Union{AbstractFloat,Integer}}) = Arf
+Base.promote_rule(::Type{Arb}, ::Type{<:Union{AbstractFloat,Integer,Arf}}) = Arb
+Base.promote_rule(
+    ::Type{Acb},
+    ::Type{<:Union{AbstractFloat,Integer,Complex,Arf,Arb,AcbRef}},
+) = Acb
+
+for (jf, af) in [(:+, :add!), (:-, :sub!), (:*, :mul!), (:/, :div!)]
+    @eval function $(Expr(:., :Base, QuoteNode(jf)))(
```

Comment:
Mhm, I remember getting an error message doing it this way, but I just double checked and this indeed works :)